src/testing.py

```diff
@@ -202,33 +202,44 @@ def execute(tester, inputs, fails):
   sys.stdout.write('\nResults:\n')
   for result in results:
     sys.stdout.write(str(result) + '\n\n')
+
   cutoff = 0.9
-  similar_expected_failures = similarity(expected_failures, cutoff)
-  for s in similar_expected_failures:
-    tests = ' '.join(s.tests)
-    if s.average >= cutoff * 100.:
-      sys.stdout.write(('\nSimilar expected failures, '
-                        'average %s%% similarity with stddev %s: '
-                        '%s\n') % (s.average, s.stddev, tests))
-      sample = [f for f in expected_failures if f.test == s.tests[0]][0]
-      sys.stdout.write('Sample failure: %s\n' % sample)
-    else:
-      sys.stdout.write(('\nUngrouped expected failures, '
-                        'average %s%% similarity with stddev %s: '
-                        '%s\n') % (s.average, s.stddev, tests))
-  similar_unexpected_failures = similarity(unexpected_failures, cutoff)
-  for s in similar_unexpected_failures:
-    tests = ' '.join(s.tests)
-    if s.average >= cutoff * 100.:
-      sys.stdout.write(('\nSimilar unexpected failures, '
-                        'average %s%% similarity with stddev %s: '
-                        '%s\n') % (s.average, s.stddev, tests))
-      sample = [f for f in unexpected_failures if f.test == s.tests[0]][0]
-      sys.stdout.write('Sample failure: %s\n' % sample)
-    else:
-      sys.stdout.write(('\nUngrouped unexpected failures, '
-                        'average %s%% similarity with stddev %s: '
-                        '%s\n') % (s.average, s.stddev, tests))
+  # Calculating similarity is pretty expensive. If more than half the tests are
+  # failing, it can take minutes, and most of them are probably failing for the
+  # same fundamental reason. Skip in that case.
+  too_many_failures = int(len(inputs) * 0.5)
```

Comment:
Yes and no. `0.5` here is mirroring the `cutoff` variable above (which isn't as co-located as I'd like because of the comment), in that it's a percentage. The way I expect this to change is to tweak that percentage, say 30% gets deemed too much, we would change it to `0.3`. The problem here is the comment, framing it as "half" instead of "50%".
(Okay realistically the only time we're changing this is to remove it completely.)
Better would be to change this to `too_much_failure_percent = 0.5`, and then recalculate off `len(inputs)` later on. Much better communicates the intent.

And really it doesn't need to be int-casted, because we only use the number as a comparison, and it just adds line noise.